src/xor_name.rs

```diff
@@ -98,10 +99,39 @@ impl XorName {
         self.cmp_distance(lhs, rhs) != Ordering::Greater
     }
 
-    // Private function exposed in fmt Debug {:?} and Display {} traits.
+    /// Return true if the name is in the interval [start, end] (both endpoints inclusive).
+    pub fn between(&self, start: &XorName, end: &XorName) -> bool {
+        self >= start && self <= end
+    }
+
+    /// Returns the value of `self` divided by `rhs`.
+    pub fn divided_by(self, rhs: u32) -> XorName {
+        XorName::from_big_uint(BigUint::from_bytes_be(&self.0) / BigUint::new(vec![rhs]))
+    }
+
+    /// Private function exposed in fmt Debug {:?} and Display {} traits.
     fn get_debug_id(&self) -> String {
         format!("{:02x}{:02x}{:02x}..", self.0[0], self.0[1], self.0[2])
     }
+
+    /// Used to construct an XorName from a `BigUint`. This will panic if `value` represents a
+    /// number greater than or equal to `2^XOR_NAME_BITS`.
+    fn from_big_uint(value: BigUint) -> XorName {
+        let little_endian_value = value.to_bytes_le();
+        assert!(little_endian_value.len() <= XOR_NAME_LEN,
+                "This BigUint value exceeds the maximum capable of being held as an XorName.");
```

Comment:
We could, but it would be an error in the way we're using it currently.  I think an `assert!` in this case is reasonable as we already get similar behaviour if we e.g. create an underflow or try to divide by 0.  If we swallow this error and return a probably invalid result, should we also do that for subtraction and division?